pkg/web/wslisten.go

```diff
@@ -0,0 +1,75 @@
+package web
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type WrappedSocketListener struct {
```

Comment:
We're already in the package `web`, so we can simplify this name.  `Listener` might be a good choice.